ibis/common/geospatial.py

```diff
@@ -35,14 +35,18 @@ def _format_geo_metadata(op, value: str, inline_metadata: bool = False) -> str:
     srid = op.args[1].srid
     geotype = op.args[1].geotype
 
-    if inline_metadata and srid:
-        value = "'SRID={};{}'".format(srid, value)
-    else:
-        value = "'{}'".format(value)
+    if inline_metadata:
+        value = "'{}{}'{}".format(
+            'SRID={};'.format(srid) if srid else '',
+            value,
+            '::{}'.format(geotype) if geotype else ''
+        )
+        return value
+    value = "'{}'".format(value)
 
     if geotype not in ('geometry', 'geography'):
-        if inline_metadata and srid:
-            return value
+        # default for not inline format
+        geotype = 'geometry'
```

Comment:
It seems strange to overwrite the `geotype` here. What other possible values of `geotype` are there that this would be necessary?

Reply:
it could be `None` ..  I can change it to check if it is `None` .. not sure why I coded this in this way .. but check if it is None it is better.

for inline format we don't need to worry about geotype. but for not inline format it uses a function to handle the literal ... so we need to decide if it is geometry or geography ... 

at this time this option (inline or not) is defined inside the backend translation

Reply:
I just changed the `if` below that and it should work better now :)